File: demo/src/main/java/com/fengshuisystem/demo/entity/Post.java

package com.fengshuisystem.demo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fengshuisystem.demo.entity.enums.Status;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Nationalized;

import java.time.Instant;
import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_id", nullable = false)
    private Integer id;
    @JsonIgnore
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinColumn
    private Account account;
    @JsonIgnore
    @NotNull
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH}, optional = false)
    @JoinColumn(name = "post_category_id", nullable = false)
    private PostCategory postCategory;
    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY, cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Package packageId;
    @JsonIgnore
    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinColumn(name = "destiny_id")
    private Destiny destiny;

    @NotNull
    @Lob
    @Column(name = "content", nullable = false)
    private String content;

    @NotNull
    @Column(name = "like_number", nullable = false)
    private Integer likeNumber;

    @NotNull
    @Column(name = "dislike_number", nullable = false)
    private Integer dislikeNumber;

    @NotNull
    @Column(name = "share_number", nullable = false)
    private Integer shareNumber;

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private Status status = Status.INACTIVE;

    @NotNull
    @Column(name = "created_date", nullable = false)
    private Instant createdDate;
    @Size(max = 255)
    @NotNull
    @Column(name = "title", nullable = false)
    private String title;

    @Size(max = 255)
    @NotNull
    @Nationalized
    @Column(name = "created_by", nullable = false)
    private String createdBy;

    @Column(name = "updated_date")
    private Instant updatedDate;

    @Size(max = 255)
    @Nationalized
    @Column(name = "updated_by")
    private String updatedBy;
    @OneToMany(mappedBy = "post", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<Comment> comments = new LinkedHashSet<>();
    @OneToMany(mappedBy = "post", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<PostImage> images = new LinkedHashSet<>();

}

File: src/models/Post.ts
import { Account } from './Account';
import { PostCategory } from './PostCategory';
import { Package } from './Package';
import { Destiny } from './Destiny';
import { Status } from './enums/Status';
import { Comment } from './Comment';
import { PostImage } from './PostImage';

export class Post {
    postId: number;
    account: Account | null;
    postCategory: PostCategory | null;
    packageId: Package | null;
    destiny: Destiny | null;
    content: string;
    likeNumber: number;
    dislikeNumber: number;
    shareNumber: number;
    status: Status;
    createdDate: Date;
    title: string;
    createdBy: string;
    updatedDate: Date | null;
    updatedBy: string | null;
    comments: Set<Comment>;
    images: Set<PostImage>;

    constructor() {
        this.postId = 0;
        this.account = null;
        this.postCategory = null;
        this.packageId = null;
        this.destiny = null;
        this.content = '';
        this.likeNumber = 0;
        this.dislikeNumber = 0;
        this.shareNumber = 0;
        this.status = Status.INACTIVE;
        this.createdDate = new Date();
        this.title = '';
        this.createdBy = '';
        this.updatedDate = null;
        this.updatedBy = null;
        this.comments = new Set<Comment>();
        this.images = new Set<PostImage>();
    }
}
